File: C++/프로그래머스/2/12985. 예상 대진표/예상 대진표.cpp
#include <iostream>
#include <cmath>

using namespace std;

int solution(int n, int a, int b)
{
    int answer = 1;
    while(min(a, b) % 2 == 0 || abs(a - b) > 1) {
        a = ceil(a / 2.0);
        b = ceil(b / 2.0);
        answer++;
    }

    return answer;
}

File: Python3/프로그래머스/2/12985. 예상 대진표/예상 대진표.py
import math

def solution(n,a,b):
    answer = 1
    while min(a, b) % 2 == 0 or abs(a - b) > 1:
        answer += 1
        a = math.ceil(a / 2)
        b = math.ceil(b / 2)
    
    return answer